app/policies/g5_authenticatable/base_policy.rb

```diff
@@ -71,16 +71,26 @@ def super_admin?
       user.present? && user.has_role?(:super_admin)
     end
 
-    def admin?
-      user.present? && user.has_role?(:admin)
+    def check_for_role(role_name,global_scope=false)
+      if user.present?
+        query = user.roles.where(name: role_name)
+        query = query.where(resource_type: nil) if global_scope
+        return query.exists?
+      else
+        return false
+      end
+    end
+
+    def admin?(global_scope=false)
```

Comment:
for those who are calling `admin?` `editor?``viewer?`without params it will have the same efect, check if there is any role with that name. Just those that call them with the params global_scope knows that they are traing to search the role but at global scope. In this case we add a new filter that is *resource_type: nil*
In other part of the code I found this
**@current_user.roles.global.exists?** where is *global* scope defined ?